apps/api/src/app/shared/usecases/create-variables-object/create-variables-object.usecase.ts

```diff
@@ -171,49 +169,28 @@ export class CreateVariablesObject {
     return _.merge(obj, val);
   }
 
-  private async getControlValues(command: CreateVariablesObjectCommand) {
-    if (command.controlValues) {
-      return [command.controlValues].flat().flatMap((obj) => Object.values(obj));
-    }
-
-    if (command.workflowId) {
-      const controls = await this.controlValuesRepository.find(
-        {
-          _environmentId: command.environmentId,
-          _organizationId: command.organizationId,
-          _workflowId: command.workflowId,
-          level: ControlValuesLevelEnum.STEP_CONTROLS,
-          controls: { $ne: null },
-        },
-        {
-          controls: 1,
-          _id: 0,
-        }
-      );
-
-      return controls
-        .map((item) => item.controls)
-        .flat()
-        .flatMap((obj) => Object.values(obj));
-    }
-
-    return [];
-  }
-
   /**
    * Extracts all variables from control values by parsing handlebars syntax {{variable}}.
    * Removes duplicates from the final result.
    *
    * @example
-   * controlValues = [ "John {{name}}", "Address {{address}} {{address}}", "nothing", 123, true ]
+   * values = [ "John {{name}}", "Address {{address}} {{address}}", "nothing", 123, true ]
    * returns = [ "name", "address" ]
    */
   @Instrument()
-  private extractAllVariables(controlValues: unknown[], isHtmlEditorEnabled: boolean): string[] {
+  private extractAllVariables({
+    controlValues,
+    isHtmlEditorEnabled,
+    variableSchema,
+  }: {
+    controlValues: unknown[];
+    isHtmlEditorEnabled: boolean;
+    variableSchema?: JSONSchemaDto;
+  }): string[] {
     const variables = controlValues.flatMap((value) => {
       const templateVariables = buildVariables({
         useNewLiquidParser: isHtmlEditorEnabled,
-        variableSchema: undefined,
+        variableSchema,
```

Comment:
fix for the issue: https://novu.slack.com/archives/C06DU7A7BPV/p1751305701191359